src/parquet/util/input.cc

```diff
@@ -235,4 +234,49 @@ void InMemoryInputStream::Advance(int64_t num_bytes) {
   offset_ += num_bytes;
 }
 
+// ----------------------------------------------------------------------
+// ChunkedInMemoryInputStream
+ChunkedInMemoryInputStream::ChunkedInMemoryInputStream(RandomAccessSource* source,
+    MemoryAllocator* pool, int64_t start, int64_t end, int64_t chunk_size) :
+    source_(source), stream_offset_(start), stream_end_(end), buffer_offset_(0) {
+  buffer_ = std::make_shared<OwnedMutableBuffer>(chunk_size, pool);
+  chunk_size_ = buffer_->size();
+  buffer_offset_ = chunk_size_;
+}
+
+const uint8_t* ChunkedInMemoryInputStream::Peek(int64_t num_to_peek, int64_t* num_bytes) {
+  *num_bytes = std::min(static_cast<int64_t>(num_to_peek), stream_end_ - stream_offset_);
+  // increase the buffer size if needed
+  if (*num_bytes > chunk_size_) {
+    buffer_->Resize(*num_bytes);
+    chunk_size_ = buffer_->size();
+    if (chunk_size_ < *num_bytes) {
+      throw ParquetException("Unable to read column data into buffer.");
```

Comment:
Should this be a DCHECK instead (seems like a "shouldn't fail" condition -- if allocation failed then `Resize` should have thrown an exception)?